yt_dlp/extractor/voicy.py

```diff
@@ -0,0 +1,145 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    ExtractorError,
+    smuggle_url,
+    traverse_obj,
+    unsmuggle_url,
+    unified_strdate,
+)
+
+import itertools
+
+
+class VoicyBaseIE(InfoExtractor):
+    # every queries are assumed to be a playlist
+    def _extract_from_playlist_data(self, value):
+        voice_id = compat_str(value['PlaylistId'])
+        upload_date = unified_strdate(value['Published'], False)
+        items = [self._extract_single_article(voice_data) for voice_data in value['VoiceData']]
+        return {
+            '_type': 'playlist',
+            'entries': items,
+            'id': voice_id,
+            'title': compat_str(value['PlaylistName']),
+            'uploader': value['SpeakerName'],
+            'uploader_id': compat_str(value['SpeakerId']),
+            'channel': value['ChannelName'],
+            'channel_id': compat_str(value['ChannelId']),
+            'upload_date': upload_date,
+        }
+
+    # NOTE: "article" in voicy = "track" in CDs = "chapter" in DVDs
+    def _extract_single_article(self, entry):
+        formats = [{
+            'url': entry['VoiceHlsFile'],
+            'format_id': 'hls',
+            'ext': 'm4a',
+            'vcodec': 'none',
+            'protocol': 'm3u8_native',
+        }, {
+            'url': entry['VoiceFile'],
+            'format_id': 'mp3',
+            'ext': 'mp3',
+            'acodec': 'mp3',
+            'vcodec': 'none',
+        }]
+        self._sort_formats(formats)
+        return {
+            'id': compat_str(entry['ArticleId']),
+            'title': entry['ArticleTitle'],
+            'description': entry['MediaName'],
+            'voice_id': compat_str(entry['VoiceId']),
+            'chapter_id': compat_str(entry['ChapterId']),
+            'formats': formats,
+        }
+
+    def _call_api(self, url, video_id, **kwargs):
+        response = self._download_json(url, video_id, **kwargs)
+        if response['Status'] != 0:
+            message = traverse_obj(response, ('Value', 'Error', 'Message'), expected_type=compat_str)
+            if not message:
+                message = 'There was a error in the response: %d' % response['Status']
+            raise ExtractorError(message, expected=False)
+        return response['Value']
+
+
+class VoicyIE(VoicyBaseIE):
```

Comment:
shouldn't this be `VoicyPlaylistIE`?

Reply:
I though it initially, but it's not "playlist", but "single podcast, split in each chapters" in the website

Reply:
then the return should be `multi_video`, no?

Reply:
that's true, I'll fix

Reply:
fixed, but not marked as outdated because there's no huge changes here. same reason for next one